crates/sui-graphql-rpc/src/consistency.rs

```diff
@@ -43,6 +43,11 @@ pub(crate) struct ConsistentNamedCursor {
     pub c: u64,
 }
 
+/// The high checkpoint watermark stamped on each GraphQL request. This is used to ensure
+/// cross-query consistency.
+#[derive(Clone, Copy)]
+pub(crate) struct CheckpointViewedAt(pub(crate) u64);
```

Comment:
nit (optional): Given the type itself is `pub(crate)`, it's fine to mark the field as `pub` (because it can't leak outside the crate anyway).